src/Altinn.App.Core/Internal/Data/DataService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Text.Json;
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Models;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Internal.Data
+{
+    /// <inheritdoc/>
+    public class DataService : IDataService
+    {
+        private readonly IDataClient _dataClient;
+        private readonly IAppMetadata _appMetadata;
+
+        private readonly JsonSerializerOptions _jsonSerializerOptions = new(JsonSerializerDefaults.Web);
+
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataService"/> class.
+        /// </summary>
+        /// <param name="dataClient"></param>
+        /// <param name="appMetadata"></param>
+        public DataService(IDataClient dataClient, IAppMetadata appMetadata)
```

Comment:
Wanted somewhere to stick logic I kept repeating for fetching and storing data through the IDataClient. Ivar mentioned he wasn't a huge fan of this DataService class. Thoughts on an alternative solution? Bit skeptical of adding all this logic to the DataClient code, as that is mostly focused on http.